src/leader.py

```diff
@@ -47,6 +48,11 @@ def __init__(self):
 
         self.fw.calibration()
 
+        self._tag_data = {
```

Comment:
You never fill this data in.

Reply:
explain yourself.

Reply:
You never modify this dictionary.

Reply:
because I put 0s there? Is there a default I should be using?

Reply:
You put the 0s there and you never change them. They should be updated every time set_speed and turn are called.